### ISSUE TITLE
Create Signup Form for Events and allow admin to turn it on and off with an event

### ISSUE THREAD

**Issue by BigActual** [realbig/RBMC_TH_STCATH#5]
From within the dashboard for an event, Contributor profile and up need to have a place to turn on event registration for individual events. We envision an on/off button within the event editor, which when turned on brings the editor the ability to check boxes for which required information will be requested of the end user, in order to complete registration. They should be able to select from Name, Phone, Email, and Address. Name is always required. The rest are check boxes. One check box to turn on the field. Another check box to require it in order to submit. Default off for all of them but the person scheduling the event should be able to choose what they need. These registrants are NOT to become "Users" for the site. In other words, NO site profile is required to register for an event.

**Comment by BigActual** [comment]
The administrator/editor needs to be able to print off the list or export a csv.

**Comment by brashrebel** [comment]
Interesting. So I am going to assume based on the details that you've requested, that the events they put on vary greatly? Like standardization of this registration form is not possible? For me, the goal to strive for would be one option for the content manager. Registration? y/n. What you've described is a need for far more control so I'm reading that the required information from users is different each time. Is that correct?

I don't mean to be difficult but I do wonder what situation they would be in where asking for basic things like name and email are not necessary. And then, if those _are_ always necessary, why should we introduce any UI for controlling their appearance? Does this make sense?

Basically, less is more. We never want to include any UI unless it is _needed_. Like really, truly needed. If it isn't, we are better serving our customer by making it just work automatically.

But still, totally up for discussion.

**Comment by BigActual** [comment]
Yes, you assume correctly and you're not being difficult. I already spoke to the team about this. Name is the only required field every time. The others become necessary in different circumstances to varied to write out here. 
This is mostly coming from the administration, that they don't want more information than they need, mostly because they have been beaten up by asking for info not needed in the past. Less information is better when not needed.

**Comment by BigActual** [comment]
I reread your question. You can streamline this by making all fields appear all the time and simplify programming by taking away the option to display the field or not. But keep the required checkboxes in play. I assume this is more what you were thinking and I'll push it through as the appropriate solution.

**Comment by brashrebel** [comment]
Sure, that sounds a little better. If we took the conversation deeper I'd be asking whether certain types of events are the ones that have certain requirements. For example, if a Youth event always requires a Phone Number field and a Dinner event always requires an email then we can even explore just basing the form on the event type. In a perfect situation, the content manager has very few questions to think about and they are all very easy. Like "Which type of these choices is this event?" Easy. But then having the user think about which fields are necessary for this event introduces a slightly more complicated decision.

I'm sure I'm taking this farther than anyone cares to. I just wish to emphasize our philosophy that everything _should_ work automatically and no UI gets introduced unless it is truly critical and no alternative exists.

**Comment by BigActual** [comment]
I can appreciate that. I don't think they have answers to those questions. Thanks for considering the details.

**Comment by BigActual** [comment]
@joelworsham Please reach out to me with questions. I'd like this to be completed by Friday please.

**Comment by joelworsham** [comment]
Hmmmmm yes this will be tricky. I can tell you the end result will not be exactly what you're asking for, but I'll make sure that the details are possible.

Standardized registration makes the most sense in all accounts, but I understand there are edge cases...

**Comment by joelworsham** [comment]
https://theeventscalendar.com/product/wordpress-event-tickets/

Would this suit?

**Comment by BigActual** [comment]
I haven't used it.

**Comment by joelworsham** [comment]
All set. I believe it's close to what you asked for. There's now a box on the right side of the screen when editing an event. It says "Event Signup". You select the form (I created one called "Event Signup Form"), then you can select which fields to show.

After submitting you may view the entries per the normal method of Gravity Forms, which is `Gravity Forms -> Entries`. The first column is "Event" which is the title of the event that submitted it.

**Comment by BigActual** [comment]
Please close if you're satisfied. I love what you've done. Great job. I'll let the client know it's available.
